assert/http_assertions.go

```diff
@@ -163,3 +164,50 @@ func HTTPBodyNotContains(t TestingT, handler http.HandlerFunc, method, url strin
 
 	return !contains
 }
+
+// HTTP asserts that a specfied handler returns set of expected values given by HttpOptions.
+//
+//	assert.HTTP(t, myHandler, "www.google.com", nil, WithCode(200), WithBody("I'm Feeling Lucky"), WithRequestHeader(http.Header{"a": []string{"b"}}, WithExpectedBody(bytes.NewBuffer("c"))))
+//
+// Returns whether the assertion was successful (true) or not (false).
+func HTTP(t TestingT, handler http.HandlerFunc, method, url string, values url.Values, options ...HttpOption) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+
+	b := &builder{}
+	for _, option := range options {
+		err := option(b)
+		if err != nil {
+			Fail(t, fmt.Sprintf("Failed to build http options, got error: %s", err))
+		}
+	}
```

Comment:
To preserve the variadic options, then the `code` cannot be set as default because there is option that would be set up always and therefore it is not an option, it would be a required parameter of function. 
Therefore I decided to make a `status.code` to be option however to make sure the usage of tests is correct, I throw a fail when no options are set as there is nothing to assert using result of `handler` call.
Basically it is a prevention of missuse of test that it does nothing except the `handler` execution.